controllers/selfnoderemediation_controller.go

```diff
@@ -142,7 +152,22 @@ func (r *SelfNodeRemediationReconciler) Reconcile(ctx context.Context, req ctrl.
 
 	if r.isStoppedByNHC(snr) {
 		r.logger.Info("SNR remediation was stopped by Node Healthcheck")
-		return ctrl.Result{}, nil
+		org := snr.DeepCopy()
+		if err := r.updateSnrProcessingCondition(remediationTerminatedByNHC, snr); err != nil {
+			return ctrl.Result{}, err
+		}
+		return ctrl.Result{}, r.patchSnrStatus(snr, org)
```

Comment:
for a later improvement: patchSnrStatus is called more and more often. I'd suggest to automatically patch the status with by deferred call, like NHC does.